src/GraphQL/Types/Scalars/EnumerationGraphType.cs

```diff
@@ -146,6 +151,21 @@ public class EnumValueDefinition : MetadataProvider
         public string Name { get; set; }
         public string Description { get; set; }
         public string DeprecationReason { get; set; }
-        public object Value { get; set; }
+        private object _value;
+        public object Value
+        {
+            get => _value;
+            set
+            {
+                _value = value;
+                if (value != null)
+                {
+                    if (value is Enum)
+                        value = Convert.ChangeType(value, Enum.GetUnderlyingType(value.GetType()));
+                }
+                UnderlyingValue = value;
+            }
+        }
+        internal object UnderlyingValue { get; set; }
```

Comment:
```suggestion
        internal object EnumUnderlyingValue { get; set; }
```

To focus that is is only for enum special cases.

Reply:
Well, it's used when comparing for all types.  It's only modified to the enum raw value for `Enum` types.  I don't mind changing it, but I'm not sure if that's any more appropriate.  Lmk if you still think we should change it; doesn't matter to me, and it's only an `internal` property anyway.

Reply:
> Well, it's used when comparing for **all** types.

I missed it. I'm OK to keep `UnderlyingValue` though it looks strange. What is _underlying_ for string or bool ?

Reply:
Agreed; I'll leave it alone for now but lmk if you think of anything more appropriate

Reply:
Let's just write a short comment and it will be enough. We have already spent too much time with enums.

Reply:
```suggestion
        /// <summary>
        /// For enums, contains the underlying enumeration value; otherwise contains <see cref="Value" />.
        /// </summary>
        internal object UnderlyingValue { get; set; }
```